universal-application-tool-0.0.1/app/views/dev/FileUploadView.java

```diff
@@ -22,43 +22,63 @@
 import models.StoredFile;
 import play.mvc.Http.Request;
 import play.twirl.api.Content;
-import services.aws.SignedS3UploadRequest;
-import services.aws.SimpleStorage;
+import services.cloud.StorageClient;
+import services.cloud.StorageServiceName;
+import services.cloud.StorageUploadRequest;
+import services.cloud.aws.SignedS3UploadRequest;
+import services.cloud.azure.BlobStorageUploadRequest;
 import views.BaseHtmlLayout;
 import views.BaseHtmlView;
 import views.HtmlBundle;
+import views.ViewUtils;
 import views.style.Styles;
 
 /** Renders a page for a developer to test uploading files. */
 public class FileUploadView extends BaseHtmlView {
+
+  private static final String AZURE_STORAGE_BLOB_CDN =
+      "https://cdn.jsdelivr.net/npm/@azure/storage-blob@10.5.0/browser/azure-storage-blob.min.js";
+
   private final BaseHtmlLayout layout;
-  private final SimpleStorage s3Client;
+  private final StorageClient storageClient;
+  private final ViewUtils viewUtils;
 
   @Inject
-  public FileUploadView(BaseHtmlLayout layout, SimpleStorage s3Client) {
+  public FileUploadView(BaseHtmlLayout layout, StorageClient storageClient, ViewUtils viewUtils) {
     this.layout = checkNotNull(layout);
-    this.s3Client = checkNotNull(s3Client);
+    this.storageClient = checkNotNull(storageClient);
+    this.viewUtils = checkNotNull(viewUtils);
   }
 
   public Content render(
       Request request,
-      SignedS3UploadRequest signedRequest,
+      StorageUploadRequest signedRequest,
       ImmutableList<StoredFile> files,
       Optional<String> maybeFlash) {
     String title = "Dev file upload";
-    HtmlBundle bundle =
-        layout
-            .getBundle()
-            .setTitle(title)
-            .addMainContent(
-                div()
-                    .with(div(maybeFlash.orElse("")))
-                    .with(h1(title))
-                    .with(div().with(fileUploadForm(signedRequest)))
-                    .with(
-                        div()
-                            .withClasses(Styles.GRID, Styles.GRID_COLS_2)
-                            .with(div().with(h2("Current Files:")).with(pre(renderFiles(files))))));
+    ContainerTag fileUploadForm;
+
+    HtmlBundle bundle = layout.getBundle();
+
+    if (signedRequest.serviceName().equals(StorageServiceName.AZURE_BLOB.getString())) {
+      fileUploadForm = azureBlobFileUploadForm((BlobStorageUploadRequest) signedRequest);
+      bundle.addFooterScripts(viewUtils.makeCdnJsTag(AZURE_STORAGE_BLOB_CDN));
```

Comment:
suggestion (blocking): please use the package manager for managing JavaScript dependencies rather than a CDN: https://github.com/Azure/azure-storage-js/blob/master/blob/README.md#npm

Reply:
issue #1763. I've updated the codebase to only inject this CDN in an Azure dev environment, PTAL.

Reply:
Updated to using a web JAR now